### PR TITLE
Clean up how player's `animation` node is set

### PR DESCRIPTION
**Pull request by mjtorn**

### CODE REVIEW

File: device/globals/player.gd [hunk 1]
```diff
@@ -345,12 +345,14 @@ func _ready():
...
-	animation = get_node("animation")
 	vm = $"/root/vm"
-	vm.register_object("player", self)
-	#_update_terrain();
+
 	if has_node("animation"):
-		get_node("animation").connect("animation_finished", self, "anim_finished")
+		animation = $"animation"
+		animation.connect("animation_finished", self, "anim_finished")
+		assert(animations)
```
Comment (fleskesvor) [inline device/globals/player.gd]: This is a good improvement. Might be better to use `vm.report_errors()` though, to provide the developer with a more helpful error message?

**Comment by mjtorn** [comment]
>+		assert(animations)
>
>This is a good improvement. Might be better to use `vm.report_errors()` though, to provide the developer with a more helpful error message?

Good one! It started off as protection against me forgetting to set the
animations, but of course it's more useful if we use the correct error
reporting.

I'll do this as soon as I have real computer time.

**Comment by mjtorn** [comment]
Is this better now?

**Comment by fleskesvor** [comment]
Looks good; thank you, @mjtorn. 🙂

**Pull request merged**